src/shared/README.txt

```diff
@@ -2,21 +2,68 @@ The "shared" directory contains the common code between the runtime and diagnost
+runtime/src/coreclr/vm/gcinfodecoder.cpp                -> diagnostics/src/shared/vm/gcinfodecoder.cpp
+runtime/src/coreclr/gc/gcdesc.h                         -> diagnostics/src/shared/gc/gcdesc.h
+runtime/src/coreclr/dlls/mscorrc/resource.h             -> diagnostics/src/shared/dlls/mscorrc/resource.h
+runtime/src/coreclr/hosts/inc/coreclrhost.h             -> diagnostics/src/shared/hosts/inc/coreclrhost.h
+runtime/src/coreclr/minipal                             -> diagnostics/src/shared/minipal
+runtime/src/coreclr/pal                                 -> diagnostics/src/shared/pal
+runtime/src/coreclr/palrt                               -> diagnostics/src/shared/palrt
+runtime/src/coreclr/utilcode                            -> diagnostics/src/shared/utilcode
+runtime/src/native/minipal                              -> diagnostics/src/shared/native/minipal
+
+Needed by dbgshim:
+
+diagnostics/src/shared/palrt/bstr.cpp
+
+HAVE_PROCFS_MAPS is needed by diagnostics/src/shared/pal/src/thread/process.cpp
+
+diagnostics/src/shared/pal/src/configure.cmake
+diagnostics/src/shared/pal/src/config.h.in
+
+There are a lot of include and source files that need to be carefully merged from the runtime because 
+there is functions that the diagnostics repo doesn't need (i.e. Mutexes) and functions that were removed
+from the runtime PAL that the diagnostics repo needs (i.e. RemoveDirectoryA).
+
+diagnostics\src\shared\inc\arrayholder.h                - needs #pragma once
+diagnostics\src\shared\inc\daccess.h                    - mostly stripped down except for macros the other shared/inc files need
+diagnostics\src\shared\inc\ex.h                         - stripped some parts of the EX_* macros out
+diagnostics\src\shared\inc\holder.h                     - needs ReleaseHolder #ifndef SOS_INCLUDE out
+diagnostics\src\shared\inc\predeftlsslot.h              - SOS needs all the old enums for older runtimes
+
+diagnostics\src\shared\utilcode\clrhost_nodependencies.cpp - remove DbgIsExecutable()
+diagnostics\src\shared\utilcode\debug.cpp
+diagnostics\src\shared\utilcode\ex.cpp
+diagnostics\src\shared\utilcode\util_nodependencies.cpp - gone except OutputDebugStringUtf8()
+diagnostics\src\shared\utilcode\hostimpl.cpp            - remove ClrSleepEx()
+diagnostics\src\shared\utilcode\pedecoder.cpp           - remove ForceRelocForDLL()
+
+diagnostics\src\shared\pal\src\debug\debug.cpp          - careful merging
+diagnostics\src\shared\pal\src\file\directory.cpp       - add RemoveDirectoryA for SOS
+diagnostics\src\shared\pal\src\handlemgr\handleapi.cpp
+
+diagnostics\src\shared\pal\src\safecrt\vsprintf.cpp     - vsprintf/vsnprintf needed for SOS %S
+diagnostics\src\shared\pal\src\safecrt\mbusafecrt.cpp   - need _safecrt_cfltcvt for vsprintf/vsnprint support
+diagnostics\src\shared\pal\src\safecrt\mbusafecrt_internal.h
+
+For swprintf/vswprintf support needed by SOS (files not part of the runtime anymore):
```

Comment:
We can follow @AaronRobinsonMSFT's footsteps from https://github.com/dotnet/runtime/pull/76771 and get rid of these (after this PR is merged).

Reply:
SOS needs this printf support on non-Windows platforms for the %S formatting char.  It prints a UTF16 string.  The standard Linux/MacOS printf doesn't (Windows does).